### PR TITLE
Connect component with render callback

### PR DESCRIPTION
**Pull request by dmmulroy**
Provide Connect component with render callback as an option alongside connect HOC 
see #14

### CODE REVIEW

**Comment by matheusml** [comment]
Hi @dmmulroy 

Could you please run this command, commit and push?

```
npm run format
```

**Comment by matheusml** [comment]
Also @dmmulroy could you please add unit tests?
If you don't know how, or have any questions, please feel free to ask!

**Comment by dmmulroy** [comment]
Just added equivalent coverage as the HOC version :+1:

**Comment by dmmulroy** [comment]
I also updated the readme to include some documentation on using the component approach as well

**Comment by matheusml** [comment]
Awesome work @dmmulroy 💯 

Some changes:
- Please bump the version to 4.1.0
- Please add this change to the CHANGELOG file
- Please remove the text from the README (I'll add this to the new API DOCs in the future, ok?)

Just to make sure, `connect` is exactly the same as before, but we have a new API with is Connect (the render props one) right?

@malbernaz what you think about this PR?

**Comment by dmmulroy** [comment]
The functionality of the HoC `connect` is the same and hasn't changed.  Under the covers it's using the Connect (it was before as well) component to render with a function rather than directly returning the Child/component to be wrapped.  Since we're supplying the render function to the Connect component for the HoC we can make sure it returns the <Child /> component as it was before.

File: src/react/components/connect.tsx [hunk 1]
```diff
@@ -3,35 +3,38 @@ import * as React from "react"
...
-        if (!shallowEqual(mapped, this.state)) {
-          this.setState(mapped)
-        }
-      }
-      render() {
-        return (
-          <Child store={this.context.store} {...this.props} {...this.state} />
-        )
-      }
+export class Connect extends React.Component<any> {
```
Comment (matheusml) [inline src/react/components/connect.tsx]: What you think about removing the <any>?
Correct me if I'm wrong, but I don't think that it's doing anything here..

File: src/react/components/connect.tsx [hunk 2]
```diff
@@ -3,35 +3,38 @@ import * as React from "react"
...
+    if (!shallowEqual(mapped, this.state)) {
+      this.setState(mapped)
     }
+  }
+  render() {
+    return this.props.children(this.state)
+  }
+}
+
+export default function connect(mapToProps) {
```
Comment (matheusml) [inline src/react/components/connect.tsx]: Also, what you think about having two files, one connect.ts and another one, Connect.ts?
In this project, we're trying to have one export per file, as a pattern.

File: src/react/components/connect.tsx (hunk 1 above)
Comment (dmmulroy) [reply src/react/components/connect.tsx]: That's being used to import the Connect component in `src/react/index.tsx`.
https://github.com/concretesolutions/redux-zero/pull/26/files#diff-85c4081ced2fbbdeeed5d03170343962R1

File: src/react/components/connect.tsx (hunk 2 above)
Comment (dmmulroy) [reply src/react/components/connect.tsx]: I actually wrote that way at first and then moved it back into connect.tsx.  I can move it back into it's own file if you'd like.

File: src/react/components/connect.tsx (hunk 1 above)
Comment (dmmulroy) [reply src/react/components/connect.tsx]: Oh wait, I think I misread.  What do you mean by removing the ?

File: src/react/components/connect.tsx (hunk 2 above)
Comment (matheusml) [reply src/react/components/connect.tsx]: Yes, please do that :)

File: src/react/components/connect.tsx (hunk 1 above)
Comment (matheusml) [reply src/react/components/connect.tsx]: I meant removing the "<any">. Sorry about that.

Comment (dmmulroy) [reply src/react/components/connect.tsx]: TypeScripts linter complains if I remove <any>, any idea how I can get past/around that?

File: src/react/components/connect.tsx (hunk 2 above)
Comment (dmmulroy) [reply src/react/components/connect.tsx]: What do you want to call the two files?  The linter complaines with two files named `Connect.tsx` and `connect.tsx`.  Would you be okay with something like `Connect.tsx` as the Component w/ render callback and `connect-hoc.tsx` or `ConnectHOC.tsx`.  This wouldn't impact the imports from the library.

File: src/react/components/connect.tsx (hunk 1 above)
Comment (matheusml) [reply src/react/components/connect.tsx]: Really? Ok, leave it there, in the future I'll try to fix it

File: src/react/components/connect.tsx (hunk 2 above)
Comment (matheusml) [reply src/react/components/connect.tsx]: Didn't know about that as well, so better to leave the way it is.

**Comment by matheusml** [comment]
For me this is ok now, I'm just waiting for @malbernaz 's opinion

**Comment by malbernaz** [comment]
Everything seems to be OK. We can merge this 👍🏼

**Comment by matheusml** [comment]
Thanks @dmmulroy 💯 💯 💯

**Comment by dmmulroy** [comment]
No problem!

**Pull request merged**